client/src/components/TreeDropdownField/TreeDropdownField.js

```diff
@@ -75,6 +91,9 @@ class TreeDropdownField extends Component {
     // No more path means this is the complete tree
     let node = this.props.tree;
     for (const next of this.props.visible) {
+      if (!node.children) {
+        break;
+      }
       node = node.children.find((child) => (child.id === next));
```

Comment:
I think that's the point of `node` though, as it is traversing down the hierarchy `this.props.tree` recursively to get to the last entry of visible
we might be able to refactor this to use the `.reduce()` function so that it is self-contained but I think it's not really worth the effort for this case